### DISCUSSION TITLE
Is it possible to filter RSS feeds by location or category?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#327] [Ask a question] [answered]
Because the blog and wiki plugins already support multiple template locations, it would make sense to also reflect this behaviour in the feed plugin. Currently you can only filter feed entries by template, but not by location. Filtering feeds by location is useful for e.g. blog sites with multiple categories where visitors should have the possibility to choose their prefered topic, beside subscribing  to all blog posts. Also some topic-related blogging "planets" require you to submit a special feed if your blog contains more topics than the desired one. 

As a personal example: I have a podcast site with two different series and would like to give my visitors the possibility to subscribe to either all episodes from all series or just the podcast they are interested in. 

Is this possible without rewriting the whole feed plugin? I wasn't successful modifying it and gave up for now. 

Steffen

**Comment by markseuffert** [comment, marked as answer]
Hi Steffen. 

I looked into the code. You can use the API to filter for meta data, for example `Tag: aetherklaenge`. Here's a code example `if($_REQUEST["tag"]) $pages->filter("tag", $_REQUEST["tag"]);`. Now your special feed should be available as `http://website/feed/tag:aetherklaenge/page:feed.xml`. 

Hope this helps... and perhaps we can add this feature to the official [feed plugin](https://github.com/annaesvensson/yellow-feed).

**Comment by schulle4u** [comment]
Thank you, I'll give it a try in the next days.

**Comment by schulle4u** [comment]
OK, got it to work as you described. I still have to find out how to modify the alternate rss subscription in site metadata, but at least I now have a more flexible feed plugin. :)

**Comment by schulle4u** [comment]
So far everything is working now. If interested, this is what I've implemented into my [Podcast plugin](https://github.com/schulle4u/yellow-plugin-podcast): 

* Filter feeds by tag, e.g. `http://website/feed/tag:news` for HTML output and `http://website/feed/tag:news/page:feed.xml` for RSS subscription.
* Display tag in page heading and as feed title if present, otherwise use the regular page title.
* Subscribe to tag-filtered feeds using alternate browser method. This should also work wherever a `tag:` parameter is present in the URL.

Feel free to adapt my idea also for the original feed plugin if this is useful. I otherwise could backport my changes and make a pull request. Thanks again for this effective one-liner. :)

Steffen

**Comment by markseuffert** [comment]
Done... added to feed plugin.

**Comment by schulle4u** [comment]
Thanks, works even better than my implementation. Think I'll rework my podcast fork as soon as world cup permits. :)

Steffen

**Comment by markseuffert** [comment]
The [feed extension](https://github.com/annaesvensson/yellow-feed) now supports filtering by author, tag or folder. Let's say you want a RSS feed for the "help" content folder, this would be available as `http://website/feed/folder:help/page:feed.xml`. More examples can be found in the documentation.

**Reply by schulle4u** [reply]
Thanks for adding more flexibility to the feed extension. For my current project workflow I'll stay with tag filtering, but Ive ported the new options into my [podcast extension](https://github.com/schulle4u/yellow-extensions-schulle4u/tree/master/podcast).
